File: MCPT/stew.py
n, k = map(int,input().split())
a = list(map(int, input().split()))
a.sort()
psa = [0 for i in range(n+1)]
for i in range(1,n+1):
    psa[i] = psa[i - 1] + a[i-1]

mnt = float("inf")
ans = 0
for i in range(1, n-k+2):
    mnc = a[i+k-2] - a[i-1]
    if mnc < mnt:
        mnt = mnc
        ans = psa[i+k-1] - psa[i-1]
    elif mnc == mnt:
        ans = psa[i+k-1] - psa[i-1]

print(ans)

File: MCPT/stew.cpp
#include <bits/stdc++.h>
using namespace std;

const int mxn = 2e5+5;

int ans = 0, n, k, a[mxn], psa[mxn];

int main() {
    scanf("%d %d",&n,&k);
    for (int i = 1; i <= n; i++) {
        scanf("%d",&a[i]);
    }
    sort(a + 1, a + n + 1);
    for (int i = 1; i <= n; i++) {
        psa[i] = psa[i - 1] + a[i];
    }
    int mnt = INT_MAX;
    int ans = 0;
    for (int i = 1; i <= n - k + 1; i++) {
        int mnc = a[i + k - 1] - a[i];
        if (mnc < mnt) {
            mnt = mnc;
            ans = psa[i + k - 1] - psa[i - 1];
        } else if (mnc == mnt) {
            ans = psa[i + k - 1] - psa[i - 1];
        }
    }
    printf("%d", ans);
}